packages/help-center/src/hooks/use-chat-status.ts

```diff
@@ -1,14 +1,19 @@
-import { isTestModeEnvironment } from '@automattic/zendesk-client';
+import { isTestModeEnvironment, useCanConnectToZendeskMessaging } from '@automattic/zendesk-client';
 import { useSupportActivity } from '../data/use-support-activity';
 import { useSupportStatus } from '../data/use-support-status';
 
 export default function useChatStatus() {
-	const { data: supportStatus } = useSupportStatus();
+	const { data: canConnectToZendesk } = useCanConnectToZendeskMessaging();
+	// No need to fetch support status if they can't reach ZD
+	const { data: supportStatus } = useSupportStatus( !! canConnectToZendesk );
 	const availability = supportStatus?.availability;
 
 	// All paying customers are eligible for chat.
 	// See: pdDR7T-1vN-p2
-	const isEligibleForChat = Boolean( supportStatus?.eligibility?.is_user_eligible );
+	// They're only eligible if they can connect to Zendesk.
+	const isEligibleForChat = Boolean(
+		supportStatus?.eligibility?.is_user_eligible && canConnectToZendesk
+	);
```

Comment:
I find it confusing that this `isEligibleForChat` condition is different from the one on the file avobe.

Reply:
When `canConnectToZendesk` is `false`, `supportStatus?.eligibility?.is_user_eligible` will be `null`, right? So maybe we can skip it.

Reply:
> I find it confusing that this isEligibleForChat condition is different from the one on the file avobe.

Yes `isEligibleForChat` became blunt (not 100% sure this is a good idea). It now encapsulates both pieces of information (`is_user_eligible` and `canConnectToZendesk`). Well, we need to know both of these in `notices` because we need to know if the user is `is_user_eligible` in separation of their ZD connectivity status.

> When canConnectToZendesk is false, supportStatus?.eligibility?.is_user_eligible will be null, right? So maybe we can skip it.

This is too confusing, I tried imagining the truth table, and it's too complicated. I'll rethink this.